FirebaseAdmin/FirebaseAdmin.IntegrationTests/FirebaseMessagingTest.cs

```diff
@@ -0,0 +1,53 @@
+// Copyright 2018, Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Xunit;
+using FirebaseAdmin.Messaging;
+
+namespace FirebaseAdmin.IntegrationTests
+{
+    public class FirebaseMessagingTest
+    {
+        public FirebaseMessagingTest()
+        {
+            IntegrationTestUtils.EnsureDefaultApp();
+        }
+
+        [Fact]
+        public async Task Send()
+        {
+            var message = new Message()
+            {
+                Topic = "foo-bar",
+                Notification = new Notification()
+                {
+                    Title = "Title",
+                    Body = "Body",
+                },
+                AndroidConfig = new AndroidConfig()
+                {
+                    Priority = Priority.Normal,
+                    TimeToLive = TimeSpan.FromHours(1),
+                    RestrictedPackageName = "com.google.firebase.testing",
+                },
+            };
+            var id = await FirebaseMessaging.DefaultInstance.SendAsync(message, dryRun: true);
```

Comment:
I don't suppose there's any way to see that the message actually got all the parts it was supposed to?

It just seems kind of odd for an "integration test" to do a `dryRun` anyway.

Reply:
dryRun mode does validate for message content (both locally and at the server). The operation will fail if the message is invalid.

Reply:
Yeah I agree with @bklimt this isn't great at the moment.  Last time @alexames looked into testing this in the client SDKs end to end there wasn't really a way to do so without spinning up a XMPP server.  If the REST request is formatted appropriately, I guess that's good enough for the mo'.

Reply:
We do check the message content in our unit tests.